lib/DDG/Meta/Data.pm

```diff
@@ -122,6 +125,29 @@ sub get_ia {
     return $m;
 }
 
+sub _satisfy {
+	my ($ia, $by, $lookup) = @_;
+	return $by->($ia, $lookup) if ref $by eq 'CODE';
+	my $pby = $ia->{$by};
+	ref $lookup eq 'CODE' ? $lookup->($pby) : $pby eq $lookup;
+}
+
+# filter_ias({ repo => 'goodies', dev_milestone => 'live'... })
+# Lookups combine as an AND operation.
+sub filter_ias {
+	my ($lookups) = @_;
+	my @results = @ia_container;
+	while (my ($by, $lookup) = each %$lookups) {
+		return () unless @results;
+		my @lookup = ref $lookup eq 'ARRAY' ? @$lookup : ($lookup);
+		@results = grep {
+			my $ia = $_;
+			any { _satisfy($ia, $by, $_) } @lookup;
```

Comment:
I think it would screw with the readability to stick it in here; are Perl calls expensive? (Annoyingly Perl won't inline non-constant subs - for 'good' reasons).